lib/osf-components/addon/components/registries/finalize-registration-modal/manager/component.ts

```diff
@@ -21,21 +22,27 @@ export interface FinalizeRegistrationModalManager {
     setEmbargoEndDate: (embargoEndDate: Date | null) => void;
     setCreateDoi: (createDoi: boolean) => void;
     submittingRegistration: boolean;
+    draftManager: DraftRegistrationManager;
 }
 
 @layout(template)
 @tagName('')
 export default class FinalizeRegistrationModalManagerComponent extends Component.extend({
     submitRegistration: task(function *(this: FinalizeRegistrationModalManagerComponent) {
         try {
+            this.draftManager.validateAllVisitedPages();
             yield this.registration.save();
 
             if (this.onSubmitRegistration) {
                 this.onSubmitRegistration(this.registration.id);
             }
         } catch (error) {
+            if (error.errors) {
```

Comment:
One last minor thing. wonder how TSLint didn't catch this 🤔 
```suggestion
            if (Array.isArray(error.errors) && error.errors.length) {
```